### PR TITLE
First implementation of returning partial fragment on timeout

### LINKED ISSUE #17: Empty fragments from `DefaultRequestHandlerModel` on low-rate inputs
When `DefaultRequestHandlerModel` gets a request for data in a time window `[start, end]`, it waits until the latency buffer has an object with time > `end`, so that it knows it has all of the data in the window (assuming the inputs are time-ordered). If an object with time > `end` does not arrive within the timeout period, the request times out and an empty fragment is returned.

This situation is very unlikely with high-rate inputs, such as WIB frames, but is much more likely with downstream objects from the DS system, such as Trigger candidates.

I think the easiest and most general way of dealing with this is to have `DefaultRequestHandlerModel` return a "partial" fragment at timeout, with whatever is in the buffer. We'll have to make sure there is some way to indicate that the fragment _may_ be incomplete (a flag in `FragmentHeader`, presumably).

Another possible approach is to have the code that feeds the buffer add periodic "fake" TCs so that the request handler will see a (fake) object with time > `end`, even if a real TC has not yet arrived. The disadvantages I see to this approach are that it requires the feeder code to know that clock rate, so it doesn't get too far ahead, and requires the feeder code to handle the case where it's fed a fake TC with time `t` and then receives a _real_ TC with time < `t`.

### PR DESCRIPTION
**Pull request by philiprodrigues**
This implements the first option described in #17 for dealing with low-rate inputs. (This PR looks like #19, which I mistakenly created against a branch that has already been merged - this PR is into develop)

### CODE REVIEW

**Comment by philiprodrigues** [comment]
There's some behaviour in `DefaultRequestHandlerModel::check_waiting_requests()` that needs discussion: at run end, any waiting requests are immediately dropped (line 506 and beyond). With low-rate objects like trigger candidates, this means the last TC fragment is always empty. For this case, I think a better behaviour would be to just wait until the request times out and return the partial fragment. The timeout will happen eventually, so we won't wait indefinitely at run stop, so I think this is safe.

I think implementing my suggestion just means removing the block in `DefaultRequestHandlerModel::check_waiting_requests()` that begins `else if (!m_run_marker.load())`

File: include/readoutlibs/concepts/RequestHandlerConcept.hpp [hunk 1]
```diff
@@ -55,13 +56,15 @@ class RequestHandlerConcept
...
     kNotYet,
+    kPartial,
     kPass,
     kCleanup,
     kUnknown
   };
   std::map<ResultCode, std::string> ResultCodeStrings{
     { ResultCode::kFound, "FOUND" },    { ResultCode::kNotFound, "NOT_FOUND" },
     { ResultCode::kTooOld, "TOO_OLD" }, { ResultCode::kNotYet, "NOT_YET_PRESENT" },
+    { ResultCode::kNotYet, "PARTIAL_RESULT" },
```
Comment (roland-sipos) [inline include/readoutlibs/concepts/RequestHandlerConcept.hpp]: Should be `kPartial`, not `kNotYet`.

**Comment by roland-sipos** [comment]
- Please find a mistake above with the enum.

- Now that the IOManager changes are on develop, could you please update the PR to reflect the changed function signatures? 

- Also, I would suggest that we test with the current changes, and we open another issue for your second comment: repond to kPartials on end of run, instead of dropping waiting requests on the floor:

> There's some behaviour in `DefaultRequestHandlerModel::check_waiting_requests()` that needs discussion: at run end, any waiting requests are immediately dropped (line 506 and beyond). With low-rate objects like trigger candidates, this means the last TC fragment is always empty. For this case, I think a better behaviour would be to just wait until the request times out and return the partial fragment. The timeout will happen eventually, so we won't wait indefinitely at run stop, so I think this is safe.
> 
> I think implementing my suggestion just means removing the block in `DefaultRequestHandlerModel::check_waiting_requests()` that begins `else if (!m_run_marker.load())`

Does this sound good for you?

**Review by roland-sipos — APPROVED** [review]
Went through it multiple times, looks good to me.

**Pull request merged**